### ISSUE TITLE
Failing reset process on lab menu screen

### ISSUE THREAD

**Issue by slartifer** [HughJeffner/FFRK-LabMem#193]
Not sure what caused labmem to start failing its reset process below.  I did restart the app/adb/etc where indicated below, and it's still failing.  It had no issues resetting yesterday; not sure if the scheduled restart could be causing this.

```
21:41:13 Battle Info
21:41:28 Crash detected!
21:41:28 Restarting FFRK
21:41:52 FFRK restarted!
21:42:06 Battle Won! (00:15) Drop: 6* Vitality Mote x1
...
21:58:48 Lab run completed! (0:39:25)
21:58:48 Restarting Lab in 60 seconds...
21:59:18 Restarting Lab in 30 seconds...
21:59:38 Restarting Lab in 10 seconds...
21:59:49 Restarting Lab
22:00:00 Not enough stamina!
22:00:02 Disabled Lab
01:35:00 Restarting FFRK due to schedule: New Schedule
01:35:00 Enabled Lab
01:35:00 Restarting FFRK
01:35:00 Could not auto-start
01:35:03 Trying to auto-start
01:35:04 Could not auto-start
01:38:04 FFRK restart timed out
01:38:44 Possible hang, attempting recovery!
01:38:44 Trying to auto-start
01:38:45 Could not auto-start
01:39:34 Hang detected!
01:39:34 Restarting FFRK
...
07:12:53 Restarting FFRK
07:12:56 Trying to auto-start
07:12:57 Could not auto-start
07:15:58 FFRK restart timed out
07:16:35 Disabled Lab
07:16:38 Enabled Lab
07:16:38 Trying to auto-start
07:16:39 Could not auto-start
07:16:52 Manually activated FFRK restart
07:16:52 Restarting FFRK
07:16:52 Restarting FFRK
07:16:55 Trying to auto-start
07:16:56 Could not auto-start
07:19:55 FFRK restart timed out
...
07:22:16 FFRK-LabMem v7.6.1-beta
07:22:16 Checking for newer releases...
07:22:16 Data logging enabled, target folder: ./Data
07:22:20 Starting proxy server on 192.168.50.117:8081 https:True
07:22:20 Setting up Lab with config: .\Config\lab.standard lens only.json
07:22:20 Device available: 127.0.0.1:21503
07:22:20 Device changed: Online:127.0.0.1:21503
07:22:20 Connected to G011A (G011A G011A)
07:22:20 Detecting proxy settings...
07:22:21 Proxy settings OK (trouble loading?, try changing port)
07:22:21 Setting up frame capture
07:22:21 Setting up input
...
07:24:27 Enabled Lab
07:24:28 Manually activated FFRK restart
07:24:28 Restarting FFRK
07:24:28 [Lab] Cancelling any running tasks
07:24:29 [Adb] matches: 0
07:24:29 Could not auto-start
07:24:29 [Lab] Kill ffrk process...
07:24:31 [Lab] Launching app
07:24:32 [Lab] Entering state: Unknown
07:24:32 [Lab] Button finding loop for 180s
07:24:32 [Adb] Finding images: 4 
07:24:32 Trying to auto-start
07:24:32 [Adb] Finding images: 3 
07:24:32 [Proxy] Blocked: abort-count.smbeat.jp
07:24:33 [Adb] matches: 0
07:24:33 Could not auto-start
07:24:33 [Adb] matches: 0
07:24:33 [Adb] Finding images: 4 
07:24:34 [Proxy] https://ffrk.denagames.com/dff/_api_create_session
07:24:34 [Proxy] https://ffrk.denagames.com/dff/splash
07:24:34 [Adb] matches: 0
07:24:34 [Proxy] https://ffrk.denagames.com/favicon.ico
07:24:35 [Adb] Finding images: 4 
07:24:36 [Adb] matches: 0
07:24:36 [Adb] Finding images: 4 
07:24:37 [Adb] matches: 0
07:24:37 [Adb] Finding images: 4 
07:24:39 [Adb] matches: 0
07:24:39 [Adb] Finding images: 4 
07:24:40 [Proxy] https://ffrk.denagames.com/dff/
07:24:40 [Proxy] Tunnel to: ffrk.static.denagames.com
07:24:40 [Proxy] Tunnel to: ffrk.static.denagames.com
07:24:40 [Proxy] Tunnel to: ffrk.static.denagames.com
07:24:40 [Adb] matches: 0
07:24:40 [Adb] Finding images: 4 
07:24:42 [Adb] matches: 0
07:24:42 [Adb] Finding images: 4 
07:24:43 [Proxy] https://ffrk.denagames.com/dff/update_user_session
07:24:43 [Adb] matches: 0
07:24:43 [Adb] Finding images: 4 
07:24:44 [Adb] matches: 1, closest: 0.9798342 [363,996]
07:24:44 [Adb] Tapping screen [Minitouch] at: [363,996]
07:24:44 [Adb] Finding images: 4 
07:24:45 [Adb] matches: 1, closest: 0.9805847 [363,996]
07:24:45 [Adb] Tapping screen [Minitouch] at: [363,996]
07:24:45 [Proxy] https://ffrk.denagames.com/dff/mission/list
07:24:45 [Adb] Finding images: 4 
07:24:45 [Proxy] https://ffrk.denagames.com/dff/mission/list
07:24:45 [Proxy] https://ffrk.denagames.com/dff/event/rotation/608/get_data
07:24:46 [Proxy] https://ffrk.denagames.com/dff/theatre/has_played_theatre?theatre_id=10361201
07:24:47 [Adb] matches: 0
07:24:47 [Proxy] https://ffrk.denagames.com/dff/event/extreme/935/get_data_for_world_list
07:24:47 [Adb] Finding images: 4 
07:24:48 [Adb] matches: 0
07:24:49 [Adb] Finding images: 4 
07:24:50 [Adb] matches: 1, closest: 0.9167734 [363,600]
07:24:50 [Adb] Tapping screen [Minitouch] at: [363,600]
07:24:50 [Adb] Finding images: 4 
07:24:52 [Adb] matches: 0
07:24:52 [Proxy] https://ffrk.denagames.com/dff/event/labyrinth/4500/get_data
07:24:52 [Adb] Finding images: 4 
07:24:54 [Adb] matches: 0
07:24:54 [Adb] Finding images: 4 
07:24:56 [Adb] matches: 0
07:24:56 [Adb] Finding images: 4 
07:24:58 [Adb] matches: 0
07:24:58 [Adb] Finding images: 4 
07:25:00 [Adb] matches: 0
07:25:00 [Adb] Finding images: 4 
07:25:01 [Adb] matches: 0
...repeated continuously
```

![image](https://user-images.githubusercontent.com/17887799/157662960-94e3efa7-35c2-4d2e-8099-1870577354af.png)

**Comment by slartifer** [comment]
Note that this was all prior to the cloud sensei announcement appearing.

**Comment by HughJeffner** [comment]
I think I understand what is going on here

The problem is that when the bot is freshly started/enabled it won't automatically enter the outpost and start the lab (if you are not currently in progress).  I did this intentionally so if you start up the bot then go and enter the lab it doesn't take over in case you needed to adjust which node or party to use.  It does this by checking if there is any previous json data.  Disabling the lab clears this data.

So in your case the lab got disabled due to not enough stamina, clearing the previous data, then the schedule attempted to restart the lab.  Since this was a 'fresh start' it never entered the outpost.  In a later attempt, you started the bot fresh and manually initiated a ffrk restart `Alt+R`.  Since no previous json data present it never entered the outpost.

I figured out a better way of handling this by checking if the bot is in a ffrk restart operation instead.  That way, it will ignore the trigger if you manually clicked on it in-game but otherwise it should always enter the outpost.
